tests/transforms/test_decompose_transform.py

```diff
@@ -0,0 +1,240 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit tests for the ``decompose`` transform"""
+
+import pytest
+
+import pennylane as qml
+import pennylane.numpy as qnp
+from pennylane.operation import Operation
+from pennylane.transforms.decompose import _operator_decomposition_gen, decompose
+
+# pylint: disable=unnecessary-lambda-assignment
+# pylint: disable=too-few-public-methods
+
+
+class NoMatOp(Operation):
+    """Dummy operation for expanding circuit."""
+
+    # pylint: disable=arguments-renamed, invalid-overridden-method
+    @property
+    def has_matrix(self):
+        return False
+
+    def decomposition(self):
+        return [qml.PauliX(self.wires), qml.PauliY(self.wires)]
+
+
+class NoMatNoDecompOp(Operation):
+    """Dummy operation for checking check_validity throws error when
+    expected."""
+
+    # pylint: disable=arguments-renamed, invalid-overridden-method
+    @property
+    def has_matrix(self):
+        return False
+
+
+class InfiniteOp(Operation):
+    """An op with an infinite decomposition."""
+
+    num_wires = 1
+
+    def decomposition(self):
+        return [InfiniteOp(*self.parameters, self.wires)]
+
+
+class TestDecompose:
+    """Unit tests for decompose function"""
+
+    gate_set_inputs = [None, "RX", ["RX"], ("RX",), {"RX"}, qml.RX, [qml.RX], (qml.RX,), {qml.RX}]
+
+    iterables_test = [
+        (
+            [qml.Hadamard(0)],
+            {qml.RX, qml.RZ},
+            [qml.RZ(qnp.pi / 2, 0), qml.RX(qnp.pi / 2, 0), qml.RZ(qnp.pi / 2, 0)],
+            None,
+        ),
+        (
+            [qml.SWAP(wires=[0, 1])],
+            {qml.CNOT},
+            [qml.CNOT([0, 1]), qml.CNOT([1, 0]), qml.CNOT([0, 1])],
+            None,
+        ),
+        (
+            [qml.Toffoli([0, 1, 2])],
+            {qml.Toffoli},
+            [qml.Toffoli([0, 1, 2])],
+            None,
+        ),
+        (
+            [qml.measurements.MidMeasureMP(0)],
+            {},
+            [qml.measurements.MidMeasureMP(0)],
+            "MidMeasureMP",
+        ),
+    ]
+
+    callables_test = [
```

Comment:
This doesn't seem to get used?

Reply:
Ah yes ... oops. I copied the line and didn't change it. 🥲